### DISCUSSION TITLE
The base processor has a top speed of >200 MHz?

### DISCUSSION THREAD

**Discussion by Sam-Vervaeck** [stnolting/neorv32#987] [Q&A]
I was wondering for a bit what would be the fastest speed I could run the NEORV32 at if it only has some basic properties turned on. So I created a little block diagram with only the timer, uart0, IMEM (16384), and DMEM (8192) turned on:

![image](https://github.com/user-attachments/assets/2a3398f1-72b3-4766-b567-27077073e520)

I am running the Genesys2 kintex-7 board which provides a diff crystal clock of 200 MHz. The clocking wizard was set to output a 200 MHz clock as well. Lo and behold, it actually implemented with a WNS of 0.174 ns. 

Then I wanted to double check if the specs of the ISA extensions in the data sheet are still applicable since they are only marked as 130 MHz max. So I turned on Zfinx. To my surprise it implemented again with a WNS of 0.176 ns this time.

What are the top speeds you guys have been running the NEORV32 at?

**Comment by stnolting** [comment]
> Then I wanted to double check if the specs of the ISA extensions in the data sheet are still applicable since they are only marked as 130 MHz max. So I turned on Zfinx. To my surprise it implemented again with a WNS of 0.176 ns this time.

That's a lot - even for a 7-series Kintex! :wink: :rocket:

> What are the top speeds you guys have been running the NEORV32 at?

I have an UltraScale+ setup here running NEORV32 at 250MHz. A lot more would be possible; the critical path is caused by the many PMP entries we are using.

I would also be interested in some ASIC results 😅
Right now, I am testing a 180nm technology running at 100MHz. Seems possible, but I'm not (yet) sure where the upper limit is.
